Modules/_sre.c

```diff
@@ -563,17 +563,36 @@ pattern_error(Py_ssize_t status)
     }
 }
 
+static int
+pattern_traverse(PatternObject *self, visitproc visit, void *arg)
+{
+    Py_VISIT(self->groupindex);
+    Py_VISIT(self->indexgroup);
+    Py_VISIT(self->pattern);
+    Py_VISIT(Py_TYPE(self));
+    return 0;
+}
+
+static int
+pattern_clear(PatternObject *self)
+{
+    if (self->weakreflist != NULL) {
+        PyObject_ClearWeakRefs((PyObject *) self);
+    }
```

Comment:
PyObject_ClearWeakRefs() is designed for dealloc function, not for clear function. I'm not sure that this change is correct. See: https://bugs.python.org/issue42972#msg394541

Reply:
Oops! I actually knew that from reading about `tp_traverse` yesterday: https://docs.python.org/3/c-api/typeobj.html#c.PyTypeObject.tp_traverse